packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts

```diff
@@ -235,9 +250,12 @@ export class BucketDeployment extends CoreConstruct {
         SystemMetadata: mapSystemMetadata(props),
         DistributionId: props.distribution?.distributionId,
         DistributionPaths: props.distributionPaths,
+        // Passing through the ARN sequences dependencees on the deployment
+        PassThroughBucketArn: props.destinationBucket.bucketArn,
```

Comment:
Can't we just use the existing `DestinationBucketName` to do this sequencing? Won't it have the same effect, without changing the customer's template?

Reply:
I'm not sure I follow. If it doesn't change the template, it won't enforce sequencing, right?

Right now, the dependencies look like this, which lead to a race condition:

```
                                       ┌────────────────────┐
                     { Ref: Bucket }   │                    │
                          ┌────────────│     Deployment     │
                          │            │                    │
┌────────────┐            │            └────────────────────┘
│            │            │                                  
│   Bucket   │◀───────────┤                                  
│            │            │                                  
└────────────┘            │            ┌────────────────────┐
                          │            │                    │
                          └────────────│   BucketConsumer   │
                     { Ref: Bucket }   │                    │
                                       └────────────────────┘
```


We want them to look like this:

```
                                              { GetAtt: [..., 'BucketArn'] }               
┌────────────┐                 ┌────────────────────┐                ┌────────────────────┐
│            │ { Ref: Bucket } │                    │                │                    │
│   Bucket   │◀────────────────│     Deployment     │◀───────────────│   BucketConsumer   │
│            │                 │                    │                │                    │
└────────────┘                 └────────────────────┘                └────────────────────┘
```

Reply:
I meant template changes related to the new `PassThroughBucketArn` property that you added, regardless of whether your customers use the new `deployedBucket ` property or not.

If you used the existing `DestinationBucketName` property for this purpose, I believe existing customers would not have their template affected.

Reply:
Oh I see. Is that good enough though? What if the bucket is in another account?

Reply:
If it's in another account, sequencing within the template with `Fn::GetAtt` is irrelevant anyway 🙂.

Reply:
That is true, using the bucket still shouldn't fail.